### ISSUE TITLE
Port of Logic Engine to .Net C#

### ISSUE THREAD

**Issue by MMJM** [chrisjpatty/flume#185]
Hi, I was interested in creating a logic engine for .Net C# and was wondering if you can point me to the code I’d have to modify for that. I’ll then have a look to see what work is required. Thanks

**Comment by chrisjpatty** [comment]
Yeah sure, the root engine is all contained in this file here: https://github.com/chrisjpatty/flume/blob/master/src/RootEngine.js

**Comment by Duinrahaic** [comment]
I've been looking at this for the past couple of days and I've been trying to figure out if the root engine is the proper way to start on this ?

From my understanding of this, root engine requires knowledge of the configuration of the system. In order to handle all the ports appropriately wouldn't I need to configure the system in a way that the context of the nodes, ports, and controls and their actions ... then pass that to the C# engine? Trying to figure out what is relevant and what is the minimal amount of data at the moment.

**Comment by shealan** [comment]
I'd love to use the output of Flume graph in a Unity C# project. Did anyone manage to get this working?

**Comment by Duinrahaic** [comment]
I did not but I'm sure that it's still possible. I believe that when I was writing this it was in the midst of transferring to typescript. Since then I've moved on to other projects so I have not returned to this project in some time.
